rules/consistent-function-scoping.js

```diff
@@ -21,11 +21,7 @@ function checkReferences(scope, parent, scopeManager) {
 		const [definition] = resolved.defs;
 
 		// Skip recursive function name
-		if (definition && definition.type === 'FunctionName' && resolved.name === definition.name.name) {
-			return false;
-		}
-
-		return isSameScope(parent, resolved.scope);
+		return definition && definition.type === 'FunctionName' && resolved.name === definition.name.name ? false : isSameScope(parent, resolved.scope);
```

Comment:
Either this should be excluded due its length, or:

```suggestion
		const condition = definition && definition.type === 'FunctionName' && resolved.name === definition.name.name;
		return condition ? false : isSameScope(parent, resolved.scope);
```

Reply:
I think it shouldn't use the ternary at all:

```suggestion
		return !(definition && definition.type === 'FunctionName' && resolved.name === definition.name.name) && isSameScope(parent, resolved.scope);
```

But it's still not readable as is. Maybe:


```suggestion
		return !(
			definition &&
			definition.type === 'FunctionName' &&
			resolved.name === definition.name.name
		) && isSameScope(parent, resolved.scope);
```

But given the length, once again, it's probably not worth the change